stb_image.h

```diff
@@ -4079,7 +4112,12 @@ static int stbi__zhuffman_decode_slowpath(stbi__zbuf *a, stbi__zhuffman *z)
 stbi_inline static int stbi__zhuffman_decode(stbi__zbuf *a, stbi__zhuffman *z)
 {
    int b,s;
-   if (a->num_bits < 16) stbi__fill_bits(a);
+   if (a->num_bits < 16) {
+      if (stbi__zeof(a)) {
+         return -1;   /* report error for unexpected end of data. */
```

Comment:
Is it possible that this is too aggressive now?  I'm using stb_image to decompress files in zip archives (long story) and I started seeing some unexpected failures after this change.  When this error condition is hit, the zbuf has 13 bits in it and it's trying to decode the last byte of the stream.  Still brushing up on huffman stuff so I might be missing something obvious.  I can scrape together a test case if that's helpful

Reply:
What if you add an stbi_fill_bits(a) after the if num_bits and before the if stbi_zeof?

Reply:
actually I think this is going to need a messier fix, since it can potentially intentionally "read past the end of file" while trying to do the "fast" huffman table lookup. ugh, need to figure out the right way to do it.